modules/registry-refactoring/src/main/java/org/apache/airavata/registry/core/repositories/appcatalog/ApplicationDeploymentRepository.java

```diff
@@ -0,0 +1,159 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.airavata.registry.core.repositories.appcatalog;
+
+import org.apache.airavata.model.appcatalog.appdeployment.*;
+import org.apache.airavata.registry.core.entities.appcatalog.*;
+import org.apache.airavata.registry.core.utils.DBConstants;
+import org.apache.airavata.registry.core.utils.ObjectMapperSingleton;
+import org.apache.airavata.registry.core.utils.QueryConstants;
+import org.apache.airavata.registry.cpi.AppCatalogException;
+import org.apache.airavata.registry.cpi.ApplicationDeployment;
+import org.dozer.Mapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ApplicationDeploymentRepository extends AppCatAbstractRepository<ApplicationDeploymentDescription, ApplicationDeploymentEntity, String> implements ApplicationDeployment {
+    private final static Logger logger = LoggerFactory.getLogger(ApplicationDeploymentRepository.class);
+
+    public ApplicationDeploymentRepository() {
+        super(ApplicationDeploymentDescription.class, ApplicationDeploymentEntity.class);
+    }
+
+    protected String saveApplicationDeploymentDescriptorData(
+            ApplicationDeploymentDescription applicationDeploymentDescription, String gatewayId) throws AppCatalogException {
+        ApplicationDeploymentEntity applicationDeploymentEntity = saveApplicationDeployment(applicationDeploymentDescription, gatewayId);
+        return applicationDeploymentEntity.getAppDeploymentId();
+    }
+
+    protected ApplicationDeploymentEntity saveApplicationDeployment(
+            ApplicationDeploymentDescription applicationDeploymentDescription, String gatewayId) throws AppCatalogException {
+        String applicationDeploymentId = applicationDeploymentDescription.getAppDeploymentId();
+        Mapper mapper = ObjectMapperSingleton.getInstance();
+        ApplicationDeploymentEntity applicationDeploymentEntity = mapper.map(applicationDeploymentDescription, ApplicationDeploymentEntity.class);
+        if (gatewayId != null)
```

Comment:
Add debug logs in each if block. Then we can easily track issues without remote debugging

Reply:
Sure @DImuthuUpe 
Do you want me to just add a message or include the object as well?

Reply:
Just print the message mentioning why that block is executing